library/src/main/java/com/otaliastudios/zoom/ZoomImageView.java

```diff
@@ -89,12 +90,50 @@ public boolean onTouchEvent(MotionEvent ev) {
     public void onUpdate(ZoomEngine helper, Matrix matrix) {
         mMatrix.set(matrix);
         setImageMatrix(mMatrix);
+
+        if (!awakenScrollBars()) {
+            invalidate();
+        }
     }
 
     @Override
     public void onIdle(ZoomEngine engine) {
     }
 
+    @Override
+    protected int computeHorizontalScrollExtent() {
+        Rect localRect = new Rect();
+        getLocalVisibleRect(localRect);
+        return localRect.width();
```

Comment:
Is there any reason why using this rather than `getWidth()` and `getHeight`? It should return the same thing almost always. Also in vertical extent & in ZoomLayout callbacks.

If there is a reason, could you cache this rect in a tempRect field? So we don't create a lot of them.

Reply:
No, I think it should also work with ```getWidth()``` and ```getHeight()```.
I don't have that much experience with custom views so the "localVisibleRect" method looked like a good fit, but I guess it can be slow.